test/version.jl

```diff
@@ -136,8 +193,27 @@ for major=0:3, minor=0:3, patch=0:3
     end
 end
 
-# julia_version.h version test
+# check_new_version
+import Base.check_new_version
+@test check_new_version([v"1", v"2"], v"3") == nothing
+@test_throws AssertionError check_new_version([v"2", v"1"], v"3")
+@test_throws ErrorException check_new_version([v"1", v"2"], v"2")
+@test check_new_version(VersionNumber[], v"0") == nothing
+@test check_new_version(VersionNumber[], v"0.0.1") == nothing
+@test_throws MethodError check_new_version(VersionNumber,[], v"0.0.2")
+@test check_new_version(VersionNumber[], v"0.1") == nothing
+@test_throws MethodError check_new_version(VersionNumber, [], v"0.2")
+@test check_new_version(VersionNumber[], v"1") == nothing
+@test_throws MethodError check_new_version(VersionNumber, [], v"2")
+@test_throws ErrorException check_new_version(VersionNumber[v"1", v"2", v"3"], v"2")
+@test_throws ErrorException check_new_version([v"1", v"2"], v"4")
+@test_throws ErrorException check_new_version([v"1", v"2"], v"2-rc")
+
+# banner
+import Base.banner
+@test banner() == nothing
```

Comment:
As @tkelman remarks this should not print to stdout, but to an internal `IOBuffer`.

```julia
io = IOBuffer()
@test banner(io) == nothing
@test length(takebuf_string(io) > 50)
```